module/VuFind/src/VuFind/Content/Summaries/Demo.php

```diff
@@ -56,6 +58,8 @@ public function loadByIsbn($key, \VuFindCode\ISBN $isbnObj)
         return [
             'Demo summary key: ' . $key,
             'Demo summary ISBN: ' . $isbnObj->get13(),
+            (new PropertyString('Demo non-HTML summary'))
+                ->setHtml('<strong>Demo HTML Summary:</strong><ul><li>Item 1</li><li>Item 2</li></ul>'),
```

Comment:
This is obviously a demo, but it assumes we have access to both HTML and non-HTML versions.  In my original use case, I have a record title that may or may not have HTML in it.  Is the intent that I would call strip_tags for version passed to the constructor, just in case?  Or is it harmless to pass the (potential) HTML version to the constructor as well as to `setHtml`?

If makes no difference, then would it be better to have `setHtml` just set a boolean flag rather than separate version of the string?

Reply:
I suppose another possible option could be to have `setHtml` set a stripped-tag version of text into the main string if that value is otherwise empty. Though maybe that's too opinionated to be a good idea...

Reply:
> I suppose another possible option could be to have setHtml set a stripped-tag version of text into the main string if that value is otherwise empty.

This would create a discrepancy with the result if the user just passed the `__html` property as part of `$properties` in the PropertyString constructor, where the extra logic setting the main string would not be called.

Reply:
Good point, that seems to support my "too opinionated to be a good idea" comment. In any case, I do think it would probably make sense to pass a tag-stripped version of text to the constructor to be on the safe side; otherwise, in a situation where HTML was present but the extended escaper was not used, the HTML tags would presumably be rendered in an undesirable way.

Reply:
> I do think it would probably make sense to pass a tag-stripped version of text to the constructor to be on the safe side; otherwise, in a situation where HTML was present but the extended escaper was not used, the HTML tags would presumably be rendered in an undesirable way.

I think that logic is sound, but in theory then we would want to strip_html all over the place in existing record drivers.  I.e. are we sure none of the EDS data fields (abstracts, etc.) may sometimes have HTML in them?

It may help me to understand how the other built-in property `__ids` would be used; what is the base string in that case?

Reply:
Maybe one approach could be a specialized constructor on the class, like `public static fromHtmlString(string $html): PropertyString` that we could use to build the objects in a standardized way...

Reply:
The specialized constructor is a good idea. In any case, the basic string version should always be plain text, if possible. At least in theory the driver can't know where and how the field is going to be displayed, and if it's e.g. part of a plain-text email or RIS export or such, it's good to have the text version available.

Reply:
Static factory method added.

Reply:
Looks good to me!